src/main/java/com/hubspot/ringleader/watcher/PersistentWatcher.java

```diff
@@ -107,20 +114,19 @@ public void run() {
   //@Override Java 5 compatibility
   public void close() throws IOException {
     if (closed.compareAndSet(false, true)) {
-      cleanup(curatorReference.getAndSet(null));
       listeners.clear();
       lastVersion.set(-1);
-      curatorTimestamp.set(0);
       executor.shutdown();
+      parent.recordClose();
```

Comment:
I might do a try/finally with the previous stuff in the try and this in the finally just to be paranoid

Reply:
👍